code/modules/mob/living/basic/slime/actions.dm

```diff
@@ -0,0 +1,146 @@
+/datum/action/innate/slime
+	check_flags = AB_CHECK_CONSCIOUS
+	button_icon = 'icons/mob/actions/actions_slime.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	///Does the ability require a specific slime lifestage?
+	var/life_stage_required
+	///Does the ability requires the slime to hit max growth?
+	var/needs_growth = FALSE
+	///Does the ability cost nutrition?
+	var/nutrition_cost = 0
+
+/datum/action/innate/slime/IsAvailable(feedback = FALSE)
+	. = ..()
+	if(!.)
+		return
+
+	var/mob/living/basic/slime/slime_owner = owner
+
+	if(!isnull(life_stage_required) && slime_owner.life_stage != life_stage_required)
+		return FALSE
+
+	if(slime_owner.nutrition < nutrition_cost)
+		return FALSE
+
+	if(!needs_growth) //always available if does not need growth
+		return TRUE
+
+	return slime_owner.amount_grown >= SLIME_EVOLUTION_THRESHOLD
+
+//Evolving
+
+/datum/action/innate/slime/evolve
+	name = "Evolve"
+	button_icon_state = "slimegrow"
+	desc = "This will let you evolve from baby to adult slime."
+	life_stage_required = SLIME_LIFE_STAGE_BABY
+	needs_growth = TRUE
+	nutrition_cost = SLIME_EVOLUTION_COST
+
+/datum/action/innate/slime/evolve/Activate()
+	var/mob/living/basic/slime/slime_owner = owner
+	slime_owner.evolve()
+
+///Turns a baby slime into an adult slime
+/mob/living/basic/slime/proc/evolve()
```

Comment:
Is this called from anywhere but this action? Or overridden by any subtypes? If not, then it can just be on the action itself rather than on the mob

Reply:
I'll do this, and I will do the same for the reproduction action.

Reply:
Whoops reproduce() is used by a crossbreed. I'll leave that alone for now, but will rework it later when that crossbreed is gone.